packages/nimble-components/src/drawer/index.ts

```diff
@@ -25,109 +24,83 @@ declare global {
+     * Closes the Drawer
+     * @param reason An optional value indicating how/why the Drawer was closed.
      */
-    public close(reason: CloseReason): void {
+    public override close(reason: CloseReason): void {
         if (!this.open || this.closing) {
             throw new Error('Drawer is not open or already closing');
         }
-        this.closeReason = reason;
-        this.closeDialog();
+        this.startClosing(reason);
     }
 
     /**
      * @internal
      */
-    public cancelHandler(event: Event): boolean {
+    public override cancelHandler(event: Event): boolean {
         // Allowing the dialog to close itself bypasses the drawer's animation logic, so we
         // should close the drawer ourselves when preventDismiss is false.
         event.preventDefault();
 
         if (!this.preventDismiss) {
-            this.closeReason = UserDismissed;
-            this.closeDialog();
+            this.startClosing(UserDismissed);
         }
         return true;
     }
 
-    private readonly animationEndHandlerFunction = (): void => this.animationEndHandler();
-
-    private openDialog(): void {
-        this.dialog.showModal();
+    protected override openDialog(): void {
+        this.dialogElement.showModal();
         this.triggerAnimation();
     }
 
-    private closeDialog(): void {
+    private readonly animationEndHandlerFunction = (): void => this.animationEndHandler();
+
+    private startClosing(reason: CloseReason | UserDismissed): void {
+        this.closeReason = reason;
         this.closing = true;
         this.triggerAnimation();
     }
 
+    private doneClosing(): void {
+        this.closing = false;
+        this.notifyClosed(this.closeReason);
+    }
+
     private triggerAnimation(): void {
-        this.dialog.classList.add('animating');
+        this.dialogElement.classList.add('animating');
         if (this.closing) {
-            this.dialog.classList.add('closing');
+            this.dialogElement.classList.add('closing');
         }
 
-        this.dialog.addEventListener(
+        this.dialogElement.addEventListener(
             eventAnimationEnd,
             this.animationEndHandlerFunction
         );
     }
 
     private animationEndHandler(): void {
-        this.dialog.removeEventListener(
+        this.dialogElement.removeEventListener(
             eventAnimationEnd,
             this.animationEndHandlerFunction
         );
-        this.dialog.classList.remove('animating');
+        this.dialogElement.classList.remove('animating');
         if (this.closing) {
-            this.dialog.classList.remove('closing');
-            this.dialog.close();
-            this.closing = false;
-            this.resolveShow!(this.closeReason);
-            this.resolveShow = undefined;
+            this.dialogElement.classList.remove('closing');
+            this.dialogElement.close();
+            this.doneClosing();
```

Comment:
It doesn't look like `doneClosing` is called anywhere but here, so I'm curious why you decided to create a helper function to wrap two lines of this function.

Reply:
It was for symmetry with the `startClosing` function that I added. One sets `closeReason` and sets `closing = true`, and the other uses `closeReason` and sets `closing = false`.